webapp/src/main/java/com/box/l10n/mojito/service/pullrun/PullRunAssetService.java

```diff
@@ -1,26 +1,172 @@
+                    Lists.partition(tuvIds, BATCH_SIZE).forEach(tuvIdsBatch -> saveTextUnitVariantsMultiRowBatch(jdbcTemplate, pullRunAsset, tuvIdsBatch, now));
+                });
+    }
+
+    /**
+     * TODO(jean) Other implementation will track translation inheritance out of the box, but not this.
+     * Though actually with current implementation the inheritance handling might be complicated. To review - how
+     * do you know fr-CA delta needs to be changed when fr-FR is changed?
+     * <p>
+     * Ran locally this was very slow.. while we expected it to be the faster option
+     */
+    @StopWatch
+    @Transactional
+    public void saveTextUnitVariantsServer(PullRunAsset pullRunAsset, List<Long> tuvIds, Long localeId) {
+        Repository repository = pullRunAsset.getPullRun().getRepository();
+        JdbcTemplate jdbcTemplate = new JdbcTemplate(dataSource);
+        meterRegistry.timer("PullRunAssetService.saveTextUnitVariantsMultiRow", Tags.of("repositoryId", Objects.toString(repository.getId())))
+                .record(() -> {
+                    // TODO(jean) might be the fastest option (if data is clean on the server side),
+                    //  inconvinient is that we over copy typically unused string
+                    // we do need some of those unused strings since pull generate localized files regardless of the
+                    // used/unused status. Maybe filter by timestamp?
+                    jdbcTemplate.update("insert into pull_run_text_unit_variant (pull_run_asset_id, tm_text_unit_variant_id, created_date) " +
+                                    "select ?, ttucv.tm_text_unit_variant_id, ? from tm_text_unit_current_variant ttucv " +
+                                    "where ttucv.asset_id = ? and ttucv.locale_id = ? ",
+                            pullRunAsset.getId(),
+                            Instant.now(),
+                            pullRunAsset.getAsset().getId(),
+                            localeId);
+                });
+    }
+
+    private void saveTextUnitVariantsBatch(PullRunAsset pullRunAsset, List<Long> tuvIds) {
+        for (Long tuvId : tuvIds) {
+            PullRunTextUnitVariant pullRunTextUnitVariant = new PullRunTextUnitVariant();
+            pullRunTextUnitVariant.setPullRunAsset(pullRunAsset);
+            pullRunTextUnitVariant.setTmTextUnitVariant(entityManager.getReference(TMTextUnitVariant.class, tuvId));
+            pullRunTextUnitVariantRepository.save(pullRunTextUnitVariant);
+        }
+        entityManager.flush();
+        entityManager.clear();
+    }
+
+    private void saveTextUnitVariantsPreparedStatementBatch(JdbcTemplate jdbcTemplate, PullRunAsset pullRunAsset, List<Long> tuvIds, Instant now) {
+        // TODO(jean) is providing the date faster?
+        jdbcTemplate.batchUpdate("insert into pull_run_text_unit_variant (pull_run_asset_id, tm_text_unit_variant_id, created_date) values (?, ?, ?)", new BatchPreparedStatementSetter() {
```

Comment:
@garionpin any feeling on passing the date vs auto generating on the server? Obviously the advantage to pass it, is to have the a single value